### PR TITLE
Bug 1798596: take into consideration all instance state for deletion

### PR DESCRIPTION
**Pull request by enxebre**
This fix deletion to take into consideration all instance state for deletion.
Otherwise instances with a provider state different than running will be leaked when deleting a machine.

### CODE REVIEW

File: pkg/actuators/machine/actuator.go [hunk 1]
```diff
@@ -347,17 +347,20 @@ func (a *Actuator) DeleteMachine(machine *machinev1.Machine) error {
 		return a.handleMachineError(machine, err, deleteEventAction)
 	}
 
-	instances, err := getRunningInstances(machine, client)
+	// Get all instances not terminated.
+	existingInstances, err := a.getMachineInstances(machine)
 	if err != nil {
-		glog.Errorf("%s: error getting running instances: %v", machine.Name, err)
+		glog.Errorf("%s: error getting existing instances: %v", machine.Name, err)
```
Comment (JoelSpeed) [inline pkg/actuators/machine/actuator.go]: How do we resolve the conflict this creates with #290? Within #290 once this is merged is probably easiest right? cc @alexander-demichev

Comment (enxebre) [reply pkg/actuators/machine/actuator.go]: thanks! let's hold the other ones and let this move forwards as it's an important bug fix that we want to back port to 4.3 right away.

**Comment by enxebre** [comment]
/approve
in behalf of https://github.com/openshift/cluster-api-provider-aws/pull/291#pullrequestreview-353870044
@JoelSpeed you need to add yourself to approvers :)

/hold cancel
because of https://github.com/openshift/cluster-api-provider-aws/pull/291#discussion_r375376171

**Pull request merged**